### PR TITLE
✨(dogwood/3/fun) add django-redis-sentinel-redux

### PR DESCRIPTION
**Pull request by rmoch**
We want to use Redis Sentinel to improve Django cache reliability,
here we add related dependencies and set default caches backend to Redis 
and MemLoc.

### CODE REVIEW

File: releases/dogwood/3/fun/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- Add django-redis-sentinel-redux to use Redis Sentinel for Django cache
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/CHANGELOG.md:14]:
```suggestion
- Add `django-redis-sentinel-redux` to use Redis Sentinel for Django cache
```

**Comment by sampaccoud** [comment]
I agree with @jmaupetit, the cache setting must be updated to remove all mentions of memcached + it would be nice if this PR can handle all existing releases at once.

**Comment by rmoch** [comment]
> it would be nice if this PR can handle all existing releases at once.

Let's agree on method on this flavor first ;-)

File: releases/dogwood/3/fun/config/cms/docker_build.py [hunk 2]
```diff
@@ -6,3 +6,12 @@
 DATABASES = {"default": {}}
 
 XQUEUE_INTERFACE = {"url": None, "django_auth": None}
+
+
+# we need available cache when building assets
+CACHES = {
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/cms/docker_build.py:12]: I don't understand what is the impact on the build of what you have done below :man_shrugging:? docker_build does not import from docker_run_production...

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 3]
```diff
@@ -161,51 +161,67 @@
 
 LOG_DIR = config("LOG_DIR", default=path("/edx/var/logs/edx"), formatter=path)
 
-MEMCACHED_HOST = config("MEMCACHED_HOST", default="memcached")
-MEMCACHED_PORT = config("MEMCACHED_PORT", default=11211, formatter=int)
+CACHE_REDIS_HOST = config("REDIS_SENTINEL_HOST", default="redis")
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/cms/docker_run_production.py:164]:
```suggestion
CACHE_REDIS_HOST = config("CACHE_REDIS_HOST", default="redis")
```

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 4]
```diff
@@ -161,51 +161,67 @@
...
+CACHE_REDIS_PORT = config("CACHE_REDIS_PORT", default=6379, formatter=int)
+CACHE_REDIS_DB = config("CACHE_REDIS_DB", default=1, formatter=int)
+CACHE_REDIS_URI = "redis://{}:{}/{}".format(CACHE_REDIS_HOST, CACHE_REDIS_PORT, CACHE_REDIS_DB)
+CACHE_REDIS_BACKEND = config("CACHE_REDIS_BACKEND", default="django_redis.cache.RedisCache")
+CACHE_REDIS_CLIENT = config("CACHE_REDIS_BACKEND", default="django_redis.client.DefaultClient")
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/cms/docker_run_production.py:169]:
```suggestion
CACHE_REDIS_CLIENT = config("CACHE_REDIS_CLIENT", default="django_redis.client.DefaultClient")
```

File: releases/dogwood/3/fun/config/lms/docker_run_production.py [hunk 5]
```diff
@@ -260,51 +260,67 @@
     # being a str()
     SESSION_COOKIE_NAME = str(config("SESSION_COOKIE_NAME"))
 
-MEMCACHED_HOST = config("MEMCACHED_HOST", default="memcached")
-MEMCACHED_PORT = config("MEMCACHED_PORT", default=11211, formatter=int)
+CACHE_REDIS_HOST = config("REDIS_SENTINEL_HOST", default="redis")
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/lms/docker_run_production.py:263]:
```suggestion
CACHE_REDIS_HOST = config("CACHE_REDIS_HOST", default="redis")
```

File: releases/dogwood/3/fun/config/lms/docker_run_production.py [hunk 6]
```diff
@@ -260,51 +260,67 @@
...
+CACHE_REDIS_PORT = config("CACHE_REDIS_PORT", default=6379, formatter=int)
+CACHE_REDIS_DB = config("CACHE_REDIS_DB", default=1, formatter=int)
+CACHE_REDIS_URI = "redis://{}:{}/{}".format(CACHE_REDIS_HOST, CACHE_REDIS_PORT, CACHE_REDIS_DB)
+CACHE_REDIS_BACKEND = config("CACHE_REDIS_BACKEND", default="django_redis.cache.RedisCache")
+CACHE_REDIS_CLIENT = config("CACHE_REDIS_BACKEND", default="django_redis.client.DefaultClient")
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/lms/docker_run_production.py:268]:
```suggestion
CACHE_REDIS_CLIENT = config("CACHE_REDIS_CLIENT", default="django_redis.client.DefaultClient")
```

File: releases/dogwood/3/fun/config/cms/docker_build.py (hunk 2 above)
Comment (rmoch) [reply releases/dogwood/3/fun/config/cms/docker_build.py:12]: This is a WIP, Cache is needed while building docker image and redis is not available. I'm testing things

Comment (rmoch) [reply releases/dogwood/3/fun/config/cms/docker_build.py:12]: would you propose a way to fix this ?

Comment (sampaccoud) [reply releases/dogwood/3/fun/config/cms/docker_build.py:12]: but the changes you make in this PR don't impact this file so if it is not working for you it should be breaking also on master. So I think the problem comes from somewhere else on your system :man_shrugging: ?

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 7]
```diff
@@ -161,51 +161,64 @@
...
+            "OPTIONS": {
+              "CLIENT_CLASS": CACHE_REDIS_CLIENT,
+            },
         },
         "general": {
-            "BACKEND": "django.core.cache.backends.memcached.MemcachedCache",
-            "LOCATION": "{}:{}".format(MEMCACHED_HOST, MEMCACHED_PORT),
-            "KEY_FUNCTION": "util.memcache.safe_key",
+            "BACKEND": CACHE_REDIS_BACKEND,
+            "LOCATION":CACHE_REDIS_URI,
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/cms/docker_run_production.py:184]:
```suggestion
            "LOCATION": CACHE_REDIS_URI,
```

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 8]
```diff
@@ -161,51 +161,64 @@
...
+            "OPTIONS": {
+              "CLIENT_CLASS": CACHE_REDIS_CLIENT,
+            },
         },
         "celery": {
-            "BACKEND": "django.core.cache.backends.memcached.MemcachedCache",
-            "LOCATION": "{}:{}".format(MEMCACHED_HOST, MEMCACHED_PORT),
-            "KEY_FUNCTION": "util.memcache.safe_key",
+            "BACKEND": CACHE_REDIS_BACKEND,
+            "LOCATION":CACHE_REDIS_URI,
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/cms/docker_run_production.py:192]:
```suggestion
            "LOCATION": CACHE_REDIS_URI,
```

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 9]
```diff
@@ -161,51 +161,64 @@
...
+            "OPTIONS": {
+              "CLIENT_CLASS": CACHE_REDIS_CLIENT,
+            },
         },
         "mongo_metadata_inheritance": {
-            "BACKEND": "django.core.cache.backends.memcached.MemcachedCache",
-            "LOCATION": "{}:{}".format(MEMCACHED_HOST, MEMCACHED_PORT),
-            "KEY_FUNCTION": "util.memcache.safe_key",
+            "BACKEND": CACHE_REDIS_BACKEND,
+            "LOCATION":CACHE_REDIS_URI,
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/cms/docker_run_production.py:200]:
```suggestion
            "LOCATION": CACHE_REDIS_URI,
```

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 10]
```diff
@@ -161,51 +161,64 @@
...
+            "OPTIONS": {
+              "CLIENT_CLASS": CACHE_REDIS_CLIENT,
+            },
         },
         "openassessment_submissions": {
-            "BACKEND": "django.core.cache.backends.memcached.MemcachedCache",
-            "LOCATION": "{}:{}".format(MEMCACHED_HOST, MEMCACHED_PORT),
-            "KEY_FUNCTION": "util.memcache.safe_key",
+            "BACKEND": CACHE_REDIS_BACKEND,
+            "LOCATION":CACHE_REDIS_URI,
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/cms/docker_run_production.py:208]:
```suggestion
            "LOCATION": CACHE_REDIS_URI,
```

**Pull request merged**